import os
from ament_index_python.packages import get_package_share_directory

from launch_ros.actions import Node
from launch import LaunchDescription, LaunchService
from launch.substitutions import LaunchConfiguration
from launch.launch_description_sources import PythonLaunchDescriptionSource
from launch.actions import IncludeLaunchDescription, DeclareLaunchArgument, OpaqueFunction

def launch_setup(context):
    compiled = os.environ['need_compile']

    if compiled == 'True':
        peripherals_package_path = get_package_share_directory('peripherals')
    else:
        peripherals_package_path = '/home/ubuntu/ros2_ws/src/peripherals'

   # camera_launch = IncludeLaunchDescription(
   #     PythonLaunchDescriptionSource(
   #         os.path.join(peripherals_package_path, 'launch/usb_cam.launch.py')),
   # )

    yolov11_detect_demo_node =  Node(
                package='yolov11_detect',
                executable='yolov11_detect_demo',
                name='yolov11_detect_demo',
                parameters=[
                    {'start': True},
                    {'image_topic': '/image_raw'},
                ],
                output='screen'
            )

    return [
            # camera_launch,
            yolov11_detect_demo_node,
            ]

def generate_launch_description():
    return LaunchDescription([
        OpaqueFunction(function = launch_setup)
    ])

if __name__ == '__main__':
    # 创建一个LaunchDescription对象(create a LaunchDescription object)
    ld = generate_launch_description()

    ls = LaunchService()
    ls.include_launch_description(ld)
    ls.run()

